plugins/search/src/apis.ts

```diff
@@ -35,12 +37,19 @@ class SearchApi {
 
   private async entities() {
     const entities = await this.catalogApi.getEntities();
-    return entities.items.map((result: any) => ({
-      name: result.metadata.name,
-      description: result.metadata.description,
-      owner: result.spec.owner,
-      kind: result.kind,
-      lifecycle: result.spec.lifecycle,
+    return entities.items.map((entity: Entity) => ({
+      name: entity.metadata.name,
+      description: entity.metadata.description || 'No description',
+      owner:
+        typeof entity.spec?.owner === 'string' ? entity.spec?.owner : 'unknown',
+      kind: entity.kind,
+      lifecycle:
+        typeof entity.spec?.lifecycle === 'string'
+          ? entity.spec?.lifecycle
+          : 'unknown',
+      url: `/catalog/${entity.metadata.namespace || ENTITY_DEFAULT_NAMESPACE}/${
+        entity.kind
+      }/${entity.metadata.name}`,
```

Comment:
OK now I'm being even more annoying ... :) But to appease @stefanalund , you may actually want to lowercase the namespace and kind. The URLs look a whole ton nicer as `/catalog/default/component/blah` than  `/catalog/default/Component/blah`. :) And it's intentionally case insensitive there.